### PR TITLE
Modify `SchedulerQueueingHints`

### PR DESCRIPTION
**Pull request by sanposhiho**
This feature gate is kind of unique one, added as Beta and enabled by default from the first step.

https://github.com/kubernetes/kubernetes/blob/master/pkg/features/kube_features.go#L777

/kind documentation
/kind bug

### CODE REVIEW

**Comment by sanposhiho** [comment]
/triage accepted

@kubernetes/sig-docs-en-owners It's kind of important since the current doc is completely wrong. PTAL. 🙏

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -701,7 +701,7 @@ Each feature gate is designed for enabling/disabling a specific feature:
 - `SELinuxMountReadWriteOncePod`: Speeds up container startup by allowing kubelet to mount volumes
   for a Pod directly with the correct SELinux label instead of changing each file on the volumes
   recursively. The initial implementation focused on ReadWriteOncePod volumes.
-- `SchedulerQueueingHints`: Enables the scheduler's _queueing hints_ enhancement,
+- `SchedulerQueueingHints`: Enables [the scheduler's _queueing hints_ enhancement](https://github.com/kubernetes/enhancements/blob/master/keps/sig-scheduling/4247-queueinghint/README.md),
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:704]: Thanks for this correction.
But, as a user, is this gate relevant to me?
How am I supposed to turn it on or off?

Comment (sanposhiho) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:704]: So, this feature is a very internal one, which normal users don't need to know about that.
But, this feature changes the critical path in the scheduling and may cause Pods being stuck in the queue in a worst-case scenario if bug there.
So, this feature gate is implemented as a safety net so that we can suggest disabling it in such cases.

Currently, we haven't found such bug so far, but the only one usecase is that, in a very large cluster, it causes a big memory usage increase in the scheduler. We'll put the mitigation in the next release though, people with such a large cluster _may_ want to disable it during v1.28.
https://github.com/kubernetes/kubernetes/issues/120622

Comment (tengqm) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:704]: @sanposhiho That is good to know. Can we leave this summary here as the explanation rather than pointing users to the lengthy KEP documentation?

Comment (sanposhiho) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:704]: I added some more context below, what do you think.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 2]
```diff
@@ -702,7 +702,7 @@ Each feature gate is designed for enabling/disabling a specific feature:
   for a Pod directly with the correct SELinux label instead of changing each file on the volumes
   recursively. The initial implementation focused on ReadWriteOncePod volumes.
 - `SchedulerQueueingHints`: Enables [the scheduler's _queueing hints_ enhancement](https://github.com/kubernetes/enhancements/blob/master/keps/sig-scheduling/4247-queueinghint/README.md),
-  which benefits to reduce the useless requeueing.
+  which benefits to reduce the useless requeueing. This feature is an internal one that most users don't need to care about. The QueueingHint feature changes the critical path of scheduling a lot and this feature gate is a safety net to go away from QueueingHint when we find some issues in it.
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:705]: Please wrap the long lines as we do for other features.
For example,

```suggestion
  which benefits to reduce the useless requeueing.
  This feature is an internal one that most users don't need to care about.
  The SchedulerQueueingHint feature changes the critical path of scheduling a lot
  and this feature gate is a safety net to go away from SchedulerQueueingHint
  when we find some issues in it.
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 3]
```diff
@@ -701,8 +701,12 @@ Each feature gate is designed for enabling/disabling a specific feature:
 - `SELinuxMountReadWriteOncePod`: Speeds up container startup by allowing kubelet to mount volumes
   for a Pod directly with the correct SELinux label instead of changing each file on the volumes
   recursively. The initial implementation focused on ReadWriteOncePod volumes.
-- `SchedulerQueueingHints`: Enables the scheduler's _queueing hints_ enhancement,
+- `SchedulerQueueingHints`: Enables [the scheduler's _queueing hints_ enhancement](https://github.com/kubernetes/enhancements/blob/master/keps/sig-scheduling/4247-queueinghint/README.md),
   which benefits to reduce the useless requeueing.
+  This feature is an internal one that most users don't need to care about.
+  The SchedulerQueueingHints feature changes the critical path of scheduling a lot
+  and this feature gate is a safety net to go away from SchedulerQueueingHints
+  when we find some issues in it.
```
Comment (alculquicondor) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:709]:
```suggestion
  The scheduler retries scheduling pods if something changes in the cluster that could make the pod scheduled.
  Queueing hints are internal signals that allow the scheduler to filter the changes in the cluster
  that are relevant to the unscheduled pod, based on previous scheduling attempts.  
```

`this feature gate is a safety net` all feature gates are safety nets, so no need to say that for this feature specifically.
